File: app/src/main/java/com/example/apktask/MidnightReceiver.java
package com.example.apktask;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

public class MidnightReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        // À minuit, on vide les tâches
        SharedPreferences prefs = context.getSharedPreferences("tasks", Context.MODE_PRIVATE);
        SharedPreferences.Editor editor = prefs.edit();
        editor.clear();  // Supprime tout
        editor.apply();

        System.out.println("Tâches supprimées à minuit !");
    }
}

File: app/src/main/java/com/example/apktask/receiver/MidnightReceiver.kt
package com.example.apktask.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.example.apktask.data.TaskRepository

/**
 * Recepteur de diffusion système — réinitialise les tâches au changement de date.
 *
 * Sécurité :
 *  - android:exported="false" dans le manifeste (seul le système peut l'invoquer)
 *  - Utilise applicationContext pour éviter les fuites de mémoire
 *  - Aucune donnée sensible n'est tracée en dehors des builds de débogage
 */
class MidnightReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action != Intent.ACTION_DATE_CHANGED) return

        // Suppression sécurisée via le dépôt (données chiffrées)
        TaskRepository(context.applicationContext).clearAll()
    }
}
